src/app/favourites/_components/FavElement.tsx

```diff
@@ -0,0 +1,86 @@
+'use client'
+
+import Image from 'next/image'
+import productImg from '../../../../public/coffee.png'
+import { IProduct } from '@/models/Products'
+
+import { useCombinedStore } from '@/store/store'
+import { useFavouritesStore } from '@/store/favStore'
+import AddToCartButton from '@/components/ui/Button'
+import ButtonHeart from '@/components/Heart/ButtonHeart'
+
+interface FavElementProps {
+  product: IProduct
+}
+
+export default function FavElement({ product }: FavElementProps) {
+  const addToCart = useCombinedStore((state) => state.add)
+  const { favouriteIds, addFavourite, removeFavourite } = useFavouritesStore()
+
+  const isFavourite =
+    favouriteIds && favouriteIds.some((favId) => favId === product.id)
```

Comment:
U do not need to calculate that propery, because all items in favourite list is favourite by default

Reply:
fixed